tests/unit/single_table/test_base.py

```diff
@@ -104,3 +108,584 @@ def test_fit(self):
+                'Data Scientist',
+                'Statistician',
+                'Computer Systems Analyst',
+                'Software Engineer',
+            ],
+            'salary': [80., 90., 50., 60., 82.]
+        })
+        conditions = {'position': 'Software Engineer', 'salary': 80.}
+        float_rtol = 0.01
+
+        # Run
+        filtered_data = BaseSynthesizer._filter_conditions(sampled, conditions, float_rtol)
+
+        # Assert
+        expected_data = pd.DataFrame({'position': ['Software Engineer'], 'salary': [80.]})
+        pd.testing.assert_frame_equal(filtered_data, expected_data)
+
+    def test__sample_rows_without_conditions(self):
+        """Test that sample rows calls ``_sample`` when conditions is ``None``."""
+        # Setup
+        data = pd.DataFrame({
+            'name': ['John', 'Doe', 'John Doe']
+        })
+        instance = Mock()
+        instance._data_processor._dtypes = pd.Series()
+        instance._data_processor.filter_valid.return_value = data
+
+        # Run
+        sampled, num_valid = BaseSynthesizer._sample_rows(instance, 3)
+
+        # Assert
+        assert num_valid == 3
+        pd.testing.assert_frame_equal(sampled, data)
+        instance._sample.assert_called_once_with(3)
+        instance._data_processor.reverse_transform.assert_called_once_with(
+            instance._sample.return_value
+        )
+        instance._data_processor.filter_valid.assert_called_once_with(
+            instance._data_processor.reverse_transform.return_value
+        )
+
+    def test__sample_rows_with_conditions(self):
+        """Test that sample rows calls with the transformed conditions the ``_sample``."""
+        # Setup
+        data = pd.DataFrame({
+            'name': ['John', 'Doe', 'John Doe'],
+            'salary': [90.0, 100.0, 80.0]
+        })
+        instance = Mock()
+        instance._data_processor._dtypes = pd.Series()
+        instance._filter_conditions.return_value = data[data.name == 'John Doe']
+        conditions = {'salary': 80.}
+        transformed_conditions = {'salary.value': 80.0}
+
+        # Run
+        sampled, num_valid = BaseSynthesizer._sample_rows(
+            instance,
+            3,
+            conditions=conditions,
+            transformed_conditions=transformed_conditions
+        )
+
+        # Assert
+        assert num_valid == 1
+        pd.testing.assert_frame_equal(sampled, data[data.name == 'John Doe'])
+        instance._sample.assert_called_once_with(3, {'salary.value': 80.0})
+        instance._data_processor.reverse_transform.assert_called_once_with(
+            instance._sample.return_value
+        )
+        instance._data_processor.filter_valid.assert_called_once_with(
+            instance._data_processor.reverse_transform.return_value
+        )
+
+    def test__sample_rows_with_previous_rows(self):
+        """Test that when calling sample rows with previous rows those are being concatenated."""
```

Comment:
minor: can we say `Test that previous rows are being concatenated when provided to `_sample`.